URBNSwiftCarousel/Classes/URBNSwCarouselTransitionController.swift

```diff
@@ -7,4 +7,341 @@
 //
 
 
+/**
+ Animation controller built to handle transitions between two images in separate view controllers.
+ By default, this class supports non-interactive transitions. To enable interactive transitions, register
+ a view for them using the registration method provided.
+ 
+ This class implements the UIViewControllerTransitioningDelegate protocol. As a consumer, you only need to implement
+ the URBNCarouselTransitioning protocol.
+ 
+ For the time being, this controller only supports transitions between images.
+ */
 
+
+public class URBNSwCarouselTransitionController: NSObject, UIViewControllerAnimatedTransitioning, UIViewControllerTransitioningDelegate, UIViewControllerInteractiveTransitioning, UIGestureRecognizerDelegate {
+    private(set) var interactive = false
+    private var viewInteractionBlocks = [UIView: URBNCarouselViewInteractionBeganClosure]()
+    private var viewPinchTransitionGestureRecognizers = [UIView: UIPinchGestureRecognizer]()
```

Comment:
same as above. I'd like to see a protocol that requires views to provide what the carousel needs. if we know what it all is here, maybe we can add it on with an extension